src/common/net.cc

```diff
@@ -163,6 +171,12 @@ namespace Pistache
         {
             return ntohs(reinterpret_cast<const struct sockaddr_in6*>(&addr_)->sin6_port);
         }
+        else if (addr_.ss_family == AF_UNIX)
+        {
+            // Ports are a meaningless concept for unix domain sockets.  Return
+            // an arbitrary value.
+            return 0;
```

Comment:
I agree about the class messiness (see my comment elsewhere).  Provided that we devise a good way to specify a unix domain address, the low-level mechanism for recording that address (a sockaddr_un) should be an implementation detail.  My first attempt at finding a way to set up a unix domain address was to piggyback on the Address::fromUnix method, as that exposed a struct sockaddr to use.  Then @win32asm and I realized that we could apply heuristics to the string Address constructor to do that job.  Perhaps the fromUnix method should be removed.

As to your original point, I have mixed feelings about throwing an exception.  As you suggested, refactoring to avoid the issue altogether is a better solution.  In the absence of that, I'm willing to go along with reviwer's consensus.  Does anyone else have an opinion here?

Reply:
To be clear: I kinda hate exceptions, but most other methods currently throw one in cases like these. But I don't have a strong preference.